lib/galaxy/webapps/galaxy/api/jobs.py

```diff
@@ -357,7 +358,13 @@ def build_for_rerun(self, trans, id, **kwd):
             raise exceptions.ObjectNotFound("Requested tool not found")
         if not tool.is_workflow_compatible:
             raise exceptions.ConfigDoesNotAllowException("Tool '%s' cannot be rerun." % (job.tool_id))
-        return tool.to_json(trans, {}, job=job)
+
+        # ensure state_inputs are formatted for rerunning with params_to_incoming
+        rerun_params = tool.to_json(trans, {}, job=job)
+        state_inputs = {}
+        params_to_incoming(state_inputs, tool.inputs, rerun_params['state_inputs'], tool.app)
+        rerun_params['state_inputs'] = state_inputs
+        return rerun_params
```

Comment:
Can you do this the other way round, have the payload in https://github.com/galaxyproject/bioblend/pull/343#issuecomment-724773705 be accepted for running tools ?

Reply:
Yep - wasn't sure which way was better

Reply:
@mvdbeek To be clear: you mean change the Galaxy tool API to accept also the format returned by `/api/jobs/{id}/build_for_rerun` , right?

Reply:
Right!